src/utils/xhr.js

```diff
@@ -91,19 +91,21 @@ export class Xhr {
    * @return {!Promise<!FetchResponse>}
    */
   fetch(input, init) {
-    // TODO (avimehta): Figure out if CORS needs be handled the way AMP does it.
     init = setupInit(init);
     return this.fetch_(input, init)
-      .then(
-        (response) => response,
-        (reason) => {
-          const targetOrigin = parseUrl(input).origin;
-          throw new Error(
-            `XHR Failed fetching (${targetOrigin}/...):`,
-            reason && reason.message
-          );
-        }
-      )
+      .catch((reason) => {
+        /*
+         * If the domain is not valid for SwG we return 404 without
+         * CORS headers and the browser throws a CORS error.
+         * We include some helpful text in the message to point the
+         * publisher towards the real problem.
+         */
+        const targetOrigin = parseUrl(input).origin;
+        throw new Error(
```

Comment:
Thanks for this!

Do you think we could add a `warn` like we do [here](https://github.com/subscriptions-project/swg-js/blob/main/src/runtime/meter-toast-api.js#L114-L120)?

We could say something like:
```
[swg.js] To fix the CORS error, please double check that ${productId} is configured correctly on https://publishercenter.google.com/ and that you have added ${currentOrigin} to your allow list.
```

Reply:
The problem is at the `fetch()` level we have no context on what exactly the request was so a more specific message is not possible to do cleanly.    Also xhr is used to fetch from publisher endpoints (AMP) and send analytics events to propensity so it's not 100% that a cors error is a publisher center config, just highly likely.

Reply:
Ok, makes sense - thanks!